### DISCUSSION TITLE
MSW messing with @apollo/client import and it's useQuery response types

### DISCUSSION THREAD

**Discussion by JonLunde** [mswjs/msw#1363] [Q&A]
Hello! I'm introducing MSW for mocking in our tests at work, but I ran into some problems and I can't find any related issues anywhere on the web.

So I thought it worth a shot to see if anyone in here have a hunch on what it could be or have experienced anything similar. I will try to explain the problems that best I can here, but it is unfortunately not possible to upload the codebase for you to experience the exact problem yourself.

The problems occur right after I've installed the msw package with yarn. If I remove the package again the problems are gone. So we know it is correlating with that.

**The problem is twofold:** 
1. `Cannot find module '@apollo/client' or its corresponding type declarations.` is popping up all over my application after installing msw.
2.  We use the Apollo client useQuery hook for fetching data all over our app. For some reason the response data type get set to 'any' causing implicit any errors all over our codebase.

Please let me know if I can provide any more information or images to help you understand my issue better.

**Comment by kettanaito** [comment]
Hey, @JonLunde. Thanks for raising this. 

From the info you've shared I can't see where MSW may interfere with this. We don't depend on any third-party request libraries, so if there's any interference it must be extremely implicit. I've also used MSW with Apollo numerous times before and haven't run into an issue like this. 

As the first thing, I advise you to rule out two things:

1. Corrupted dependencies.
2. TypeScript configuration that may cause this.

Clear Yarn cache, reinstall all the dependencies anew.

As for TypeScript,  "cannot find X type declarations" may mean that your `tsconfig.json` doesn't `include` the file where you're importing `@apollo/client` (i.e. you've excluded `*.test.js` which tries do use Apollo during tests). There may be other reasons for this as well, like custom `types` or `typeRoots` that may conflict.

> For some reason the response data type get set to 'any' causing implicit any errors all over our codebase.

This comes from the 1st issue: TypeScript in your project cannot find `@apollo/client`. So, naturally, without type definitions its signatures will be `any`. 

I'm afraid that's as far as I can advise without a reproduction repo. One easy way to rule things out is to create a clean TypeScript repo and install just Apollo and MSW there. Does the issue happen in that case? If not, the root cause is highly likely to be in your project and its config.

**Reply by JonLunde** [reply]
Thank you @kettanaito for taking the time to answer me here. After debugging for a while with no result I ended up moving to Apollo Mock. Will keep MSW in mind for another project in the future.
